pkg/cmd/server/api/validation/oauth.go

```diff
@@ -160,6 +164,18 @@ func ValidateLDAPIdentityProvider(provider *api.LDAPPasswordIdentityProvider) Va
 	return validationResults
 }
 
+// RemoteConnection fields validated separately -- this is for keystone-specific validation
+func ValidateKeystoneIdentityProvider(provider *api.KeystonePasswordIdentityProvider, identityProvider api.IdentityProvider) ValidationResults {
+	validationResults := ValidationResults{}
+
+	if !strings.HasPrefix(provider.RemoteConnectionInfo.URL, "https") {
+		validationResults.AddWarnings(fielderrors.NewFieldInvalid("provider.url", provider.RemoteConnectionInfo.URL, "Auth URL should be secure and start with https"))
+		return validationResults
```

Comment:
Making the explicit AuthenticateV3 call is the way to go. This forces v3 even if the config URL omits the version.